src/Type/StringType.php

```diff
@@ -98,19 +98,25 @@ public function accepts(Type $type, bool $strictTypes): TrinaryLogic
 			return $type->isAcceptedBy($this, $strictTypes);
 		}
 
-		if ($type instanceof TypeWithClassName && !$strictTypes) {
-			$reflectionProvider = ReflectionProviderStaticAccessor::getInstance();
-			if (!$reflectionProvider->hasClass($type->getClassName())) {
-				return TrinaryLogic::createNo();
-			}
-
-			$typeClass = $reflectionProvider->getClass($type->getClassName());
-			return TrinaryLogic::createFromBoolean(
-				$typeClass->hasNativeMethod('__toString'),
-			);
+		$thatClassNames = $type->getObjectClassNames();
+		if ($thatClassNames === [] || $strictTypes) {
+			return TrinaryLogic::createNo();
 		}
 
-		return TrinaryLogic::createNo();
+		return TrinaryLogic::createNo()->lazyOr(
```

Comment:
sorry if my question is dump,.. I don't foresee the overall change of this PR .. 

I am wondering, whether this change here means, that the new logic accepts a union of types with a `__toString` method, which it did not accept before...?
(in other words, does this change add a feature which needs a new test?)

Reply:
@ondrejmirtes what do you think - I just took a second look at this. some of my adaptions where I tried to handle cases where `$type->getObjectClassNames()` would return more than one entry, are adapted in the same fashion using TrinaryLogin::lazy*(). BUT, AFAIK so far only `UnionType` and `IntersectionType` would return more than one. And e.g. here further up above `CompoundType` is already handled, so this code is basically useless.

I could simplify it again to e.g.
- throw a `ShouldNotHappenException` in case it's more than one and handle it if it's one or zero entries
- always only handle the first entry which would be set or null

not sure if my adaptions like this one make sense. I can't test them right now for sure :/

Reply:
nevermind, cleaning it up already, I'll mark it as ready to review when I'm done. no need to check now